src/runtime/components/App.vue

```diff
@@ -37,12 +36,12 @@ const configProviderProps = useForwardProps(reactivePick(props, 'scrollBody'))
 const tooltipProps = toRef(() => props.tooltip)
 const toasterProps = toRef(() => props.toaster)
 
-const locale = computed(() => props.locale || en)
+const locale = toRef(() => props.locale)
```

Comment:
is `toRef` here necessary? `locale` is already reactive (since it is a prop)

Reply:
Well, its because the `localeContextInjectionKey` accepts ref type, you can see it at: https://github.com/nuxt/ui/blob/e7b69b7d6f0ebb3c578b9f58bcddf8ad36e6c6ce/src/runtime/composables/useLocale.ts#L8
And about replacement from `computed` to `toRef` just only for consistency like the code above (`tooltipProps` and `toasterProps`), or should I revert to the `computed` then?

Reply:
@benjamincanac I'll see what the problem might be today or tomorrow and I'll report back

Reply:
I think we should keep the `toRef` on `App` component since it works as expected, similar to what's described in the PR :thinking:

Reply:
@benjamincanac, I've put together a small demo with the current PR code. Everything works as it should. I think the PR can be approved in its current state.

https://github.com/user-attachments/assets/794252d5-0fac-4bf3-b9a7-11bc38763837

```vue
<script setup lang="ts">
import { en, ar } from '@nuxt/ui/locale'

const lang = ref(en)

onMounted(() => {
  setInterval(() => lang.value = lang.value.code === 'en' ? ar : en , 2000)
})
</script>

<template>
  <UApp :locale="lang" />
</template>
```